client/my-sites/domains/domain-management/dataviews/use-actions.tsx

```diff
@@ -0,0 +1,259 @@
+import { FEATURE_SET_PRIMARY_CUSTOM_DOMAIN } from '@automattic/calypso-products';
+import { PartialDomainData, SiteDetails } from '@automattic/data-stores';
+import { canSetAsPrimary } from '@automattic/domains-table/src/utils/can-set-as-primary';
+import {
+	type as domainTypes,
+	transferStatus,
+	useMyDomainInputMode,
+} from '@automattic/domains-table/src/utils/constants';
+import { isFreeUrlDomainName } from '@automattic/domains-table/src/utils/is-free-url-domain-name';
+import { isDomainInGracePeriod } from '@automattic/domains-table/src/utils/is-in-grace-period';
+import { isRecentlyRegistered } from '@automattic/domains-table/src/utils/is-recently-registered';
+import { isDomainRenewable } from '@automattic/domains-table/src/utils/is-renewable';
+import { isDomainUpdateable } from '@automattic/domains-table/src/utils/is-updateable';
+import {
+	domainManagementDNS,
+	domainManagementLink,
+	domainManagementTransferToOtherSiteLink,
+	domainUseMyDomain,
+} from '@automattic/domains-table/src/utils/paths';
+import { shouldUpgradeToMakeDomainPrimary } from '@automattic/domains-table/src/utils/should-upgrade-to-make-domain-primary';
+import { Action } from '@wordpress/dataviews';
+import { Icon, drawerLeft, info, update } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { navigate } from 'calypso/lib/navigate';
+import { domainManagementEditContactInfo } from '../../paths';
+import { AutoRenewDiolog } from './components/auto-renew-dialog';
+import { useDomainsDataViewsContext } from './use-context';
+
+export function useActions( { sidebarMode }: { sidebarMode?: boolean }, onClose?: () => void ) {
+	const translate = useTranslate();
+	const {
+		getFullDomain,
+		sites,
+		getSiteSlug,
+		isAllSitesView,
+		domainStatusPurchaseActions,
+		updatingDomain,
+		userCanSetPrimaryDomains = false,
+		onDomainAction,
+		handleUpdateContactInfo,
+	} = useDomainsDataViewsContext();
+
+	const actions: Action< PartialDomainData >[] = [
+		{
+			id: 'manage-domain',
+			isPrimary: true,
+			icon: drawerLeft,
+			label: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				return domain && domain.type === domainTypes.TRANSFER
+					? translate( 'View transfer' )
+					: translate( 'View settings' );
+			},
+			callback: ( domains: Array< PartialDomainData > ) => {
+				const domain = domains.length > 0 && domains[ 0 ] && getFullDomain( domains[ 0 ] );
+				if ( ! domain ) {
+					return;
+				}
+				const url = domainManagementLink( domain, getSiteSlug( domains[ 0 ] ), isAllSitesView );
+				navigate( url );
+			},
+			isEligible( partialDomain: PartialDomainData ) {
+				const domain = getFullDomain( partialDomain );
+				if ( ! domain ) {
+					return false;
+				}
+				return domain.type !== domainTypes.WPCOM;
+			},
+		},
+		{
+			id: 'manage-dns-settings',
```

Comment:
For the same domain, in production, this action opens the list view with DNS section, e.g.: `https://wordpress.com/domains/manage/all/overview/<DOMAIN>/dns/<OTHER_DOMAIN>` while this PR opens calypsofied wp-admin (e.g.: `http://calypso.localhost:3000/home/<DOMAIN>`).

Why the difference?

Reply:
It is a consequence of multiple conflict merges and unaligned changes (this PR lives too long).
Fixed here: https://github.com/Automattic/wp-calypso/pull/97677/commits/8340320081bd5e4264d5ed81d52459489add39b2